core/src/test/java/org/elasticsearch/index/query/SimpleQueryStringBuilderTest.java

```diff
@@ -260,7 +240,83 @@ public void testDefaultFieldParsing() throws IOException {
+     * checks only for our parsing extensions.
+     * 
+     * Also this relies on {@link SimpleQueryStringTests} to test most of the
+     * actual functionality of query parsing.
+     */
+    @Override
+    protected void doAssertLuceneQuery(SimpleQueryStringBuilder queryBuilder, Query query, QueryShardContext context) throws IOException {
+        assertThat(query, notNullValue());
+
+        if (queryBuilder.fields().size() > 1 && (!"".equals(queryBuilder.value()))) {
+            assertTrue("Query should have been BooleanQuery but was " + query.getClass().getName(), query instanceof BooleanQuery);
+
+            BooleanQuery boolQuery = (BooleanQuery) query;
+            if (queryBuilder.lowercaseExpandedTerms()) {
+                for (BooleanClause clause : boolQuery.clauses()) {
+                    if (clause.getQuery() instanceof TermQuery) {
+                        TermQuery inner = (TermQuery) clause.getQuery();
+                        assertThat(inner.getTerm().bytes().toString(), is(inner.getTerm().bytes().toString().toLowerCase()));
+                    }
+                }
+            }
+
+            assertThat(boolQuery.clauses().size(), equalTo(queryBuilder.fields().size()));
+            Iterator<String> fields = queryBuilder.fields().keySet().iterator();
+            for (BooleanClause booleanClause : boolQuery) {
+                assertThat(booleanClause.getQuery(), instanceOf(TermQuery.class));
+                TermQuery termQuery = (TermQuery) booleanClause.getQuery();
+                assertThat(termQuery.getTerm(), equalTo(new Term(fields.next(), queryBuilder.value().toLowerCase(Locale.ROOT))));
+            }
+
+            if (queryBuilder.minimumShouldMatch() != null) {
+                if ("1".equals(queryBuilder.minimumShouldMatch())) {
+                    assertThat(boolQuery.getMinimumNumberShouldMatch(), greaterThan(0));
+                } else if ("-1".equals(queryBuilder.minimumShouldMatch())) {
+                    assertThat(boolQuery.getMinimumNumberShouldMatch(), greaterThan(0));
+                } else if ("75%".equals(queryBuilder.minimumShouldMatch())) {
+                    assertThat(boolQuery.getMinimumNumberShouldMatch(), greaterThan(0));
+                } else if ("-25%".equals(queryBuilder.minimumShouldMatch())) {
+                    assertThat(boolQuery.getMinimumNumberShouldMatch(), greaterThan(0));
+                } else if ("2<75%".equals(queryBuilder.minimumShouldMatch())) {
+                    if (shouldClauses(boolQuery) > 2) {
+                        assertThat(boolQuery.getMinimumNumberShouldMatch(), greaterThan(0));
+                    }
+                } else if ("2<-25%".equals(queryBuilder.minimumShouldMatch())) {
+                    if (shouldClauses(boolQuery) > 2) {
+                        assertThat(boolQuery.getMinimumNumberShouldMatch(), greaterThan(0));
+                    }
+                }
```

Comment:
From first glance to me its not clear why all these assertions are the same. When is this not the case and might it be easier to just test those cases? Not sure because I don't know how the resolution works thought.